File: server/batch-service/src/main/java/com/c209/batchservice/domain/catalog/entity/Schedule.java
/**
 * This code was COPIED from "com.c209.catalog" package.
 */
package com.c209.batchservice.domain.catalog.entity;

import jakarta.persistence.*;
import lombok.Getter;

import java.time.LocalDateTime;

import static jakarta.persistence.GenerationType.IDENTITY;

@Getter
@Entity
public class Schedule {
    @GeneratedValue(strategy = IDENTITY)
    @Column(name = "schedule_id")
    @Id
    private Long id;
    private LocalDateTime startTime;
    private LocalDateTime endTime;

    @ManyToOne
    @JoinColumn(name = "performance_id")
    private Performance performance;
}

File: app/frontend/src/utils/realm/schema/Schedule.ts
import {Realm} from '@realm/react';
import Performance from './Performance.ts';

class Schedule extends Realm.Object<Schedule> {
  schedule_id!: string;
  performance_id!: string;
  start_time!: string;
  end_time!: string;

  static schema: Realm.ObjectSchema = {
    name: 'Schedule',
    properties: {
      schedule_id: 'string',
      performance_id: 'string',
      start_time: 'string',
      end_time: 'string',
    },
    primaryKey: 'schedule_id',
  };
}

export default Schedule;
